### ISSUE TITLE
Not able to reach RDB instance in a virtual network using public gateway and nat configuration

### ISSUE THREAD

**Issue by jeannot-ranaivosoa** [scaleway/terraform-provider-scaleway#2004]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.4.6

### Affected Resource(s)

* scaleway_rdb_instance
* scaleway_vpc_private_network
* scaleway_vpc_public_gateway_pat_rule
* scaleway_vpc_public_gateway_dhcp

### Terraform Configuration Files

What i tried was an application of rdb creation with dhcp support, private network and public gateway. 
It is the same snippet of code as the exemple provided in documentation here : https://registry.terraform.io/providers/scaleway/scaleway/latest/docs/resources/rdb_instance#example-with-private-network-and-dhcp-configuration

```hcl
resource "scaleway_vpc_private_network" "pn02" {
  name = "my_private_network"
}

resource "scaleway_vpc_public_gateway_dhcp" "main" {
  subnet = "192.168.1.0/24"
}

resource "scaleway_vpc_public_gateway_ip" "main" {
}

resource "scaleway_vpc_public_gateway" "main" {
  name  = "foobar"
  type  = "VPC-GW-S"
  ip_id = scaleway_vpc_public_gateway_ip.main.id
}

resource "scaleway_vpc_public_gateway_pat_rule" "main" {
  gateway_id   = scaleway_vpc_public_gateway.main.id
  private_ip   = scaleway_vpc_public_gateway_dhcp.main.address
  private_port = scaleway_rdb_instance.main.private_network.0.port
  public_port  = 42
  protocol     = "both"
  depends_on   = [scaleway_vpc_gateway_network.main, scaleway_vpc_private_network.pn02]
}

resource "scaleway_vpc_gateway_network" "main" {
  gateway_id         = scaleway_vpc_public_gateway.main.id
  private_network_id = scaleway_vpc_private_network.pn02.id
  dhcp_id            = scaleway_vpc_public_gateway_dhcp.main.id
  cleanup_dhcp       = true
  enable_masquerade  = true
  depends_on         = [scaleway_vpc_public_gateway_ip.main, scaleway_vpc_private_network.pn02]
}

resource "scaleway_rdb_instance" "main" {
  name              = "test-rdb"
  node_type         = "db-dev-s"
  engine            = "PostgreSQL-11"
  is_ha_cluster     = false
  disable_backup    = true
  user_name         = "my_initial_user"
  password          = "thiZ_is_v&ry_s3cret"
  region            = "fr-par"
  tags              = ["terraform-test", "scaleway_rdb_instance", "volume", "rdb_pn"]
  volume_type       = "bssd"
  volume_size_in_gb = 10
  private_network {
    ip_net = "192.168.1.254/24" #pool high
    pn_id  = scaleway_vpc_private_network.pn02.id
  }
}
```

### Debug Output

trying to connect to the database using the public gateway ip and public port binded to the rdb private port 
```
psql -h $PUBLIC_GATEWAY_IP --port 42 -d rdb -U username
```
### Actual Behavior

I get a connection refused error
psql: error: connection to server at "$PUBLIC_GATEWAY_IP", port 42 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
	
### Steps to Reproduce

Copy and paste the snippet of code and do a terraform apply.
After that just try to connect to the database. 

Kind regards 

1. Copy and paste the snippet of code into your ide 
2. `terraform apply` to scaleway

It could just be me not doing the right command to connect to db or doing a wrong configuration. If so i would be more that welcome to be guided to the right direction :) 

### References

* #0000

**Comment by jeannot-ranaivosoa** [comment]
Hi, 
Any update on this ? is the command wrong or perhaps the network configuration incorrect ?

**Comment by Monitob** [comment]
Hello @jeannot-ranaivosoa, The DHCP is primarily used for dynamically assigning IP addresses and network configuration parameters to devices on the gateway network. It is not designed or intended for managing databases. Therefore, attempting to parameterize a managed database through the DHCP service would not be supported.  

Instead, you should consider utilizing an IPAM service to manage their IP address assignments and configurations or assign your custom private network.  I've update the examples and other fixes on the PR #2022
